crates/relayer-framework/src/base/chain/traits/types/ibc.rs

```diff
@@ -0,0 +1,97 @@
+use crate::base::chain::traits::types::chain::HasChainTypes;
+use crate::base::core::traits::sync::Async;
+use crate::std_prelude::*;
+
+/**
+   The abstract types for a chain context when it is used for IBC
+   communication with a `Counterparty` chain context.
+
+   In contrast to [`HasChainTypes`], this trait is parameterized by a
+   `Counterparty` chain context, which is also required to implement
+   [`HasChainTypes`].
```

Comment:
Not sure if there's a typo here or not, but this is confusing: I read it as saying that `HasChainTypes` is _not_ parameterized by a `Counterparty` chain context, but that `HasChainTypes` requires a `Counterparty` chain context in order to be implemented.

Reply:
I rephrased it to two separate sentences to make it clear that it is `Counterparty` that is required to implement `HasChainTypes`.